### DISCUSSION TITLE
The system that you're working with cannot use email.

### DISCUSSION THREAD

**Discussion by yuyake** [codeigniter4/shield#964] [Q&A]
I have implemented a service using the Shield package. I need a feature where administrators can register users. Additionally, is it possible to deliver password change notifications via phone messages instead of email? The system cannot use email. It's crucial, so could you provide some hints, please?

**Comment by datamweb** [comment]
> I need a feature where administrators can register users.

You can do this using a codeigniter [custom controller](https://codeigniter.com/user_guide/incoming/controllers.html) and [Shield document descriptions](https://codeigniter4.github.io/shield/user_management/managing_users/#creating-users).

> is it possible to deliver password change notifications via phone messages instead of email?

If you mean SMS and magic link section, yes. You need to remove the magic link and write your own custom controller.
If you mean two-step login, yes, you need to [write a custom action](https://codeigniter4.github.io/shield/references/authentication/auth_actions/).

**Reply by yuyake** [reply]
Thank you for your response. I spent a considerable amount of time today trying the document you suggested.

However, the issue of not being able to log in persists. The process involves using email and an employee number.

So, after looking for the code where the administrator changes the password after user registration, it seems that it only works for their own account. I used the following code:

```php
$users = auth()->getProvider();
$user = $users->findById($post['id']);

$user->fill([
    'username' => $username,
    'email' => $email,
    'phone_number' => $post['phone_number'],
    'password' => $post['password']
]);
$users->save($user);
```

**Reply by datamweb** [reply]
> However, the issue of not being able to log in persists.

what do you mean?

These issues are really outside the Shield. The fact is that you should learn to work with the framework well before using the Shield.

Please share your minimum code. Minimum code means https://stackoverflow.com/help/minimal-reproducible-example
including controller, view, migration, and...

I have to run your code only with copy/paste, otherwise I don't have time to review.

**Reply by yuyake** [reply]
```php
$username = random_string('alnum', 16);
        $email = $username.'@random.io';
        $users = auth()->getProvider();
        $user = new User([
            'username' => $username,
            'email'    => $email,
            'phone_number' => $post['phone_number'],
            'password' => $post['password'],
        ]);
$users->save($user);
```

Config/Auth.php  
 ```php
 public array $validFields = [
         'phone_number',
 ];
```

This is the actual code used by the administrator for user registration. The field used for login is `phone_number`, as defined in `validFields`. When attempting to log in, it prompts to verify the password.

Accounts registered by users can successfully log in.

**Reply by kenjis** [reply]
Did you follow the user guide?
https://codeigniter4.github.io/shield/customization/login_identifier/
